opensrp-app/src/main/java/org/smartregister/util/JsonFormUtils.java

```diff
@@ -408,39 +404,40 @@ private static JSONObject getCheckBoxJsonObjects(Event event, JSONObject parentO
 
     public static void addObservation(Event e, JSONObject jsonObject) {
         String value = getString(jsonObject, VALUE);
+        if (StringUtils.isBlank(value)) { return; }
+
         String type = getString(jsonObject, AllConstants.TYPE);
-        String entity = CONCEPT;
-        boolean combineCheckboxOptionValues = jsonObject.optBoolean(COMBINE_CHECKBOX_OPTION_VALUES);
-        if (StringUtils.isNotBlank(value)) {
-            if (AllConstants.CHECK_BOX.equals(type)) {
-                try {
-                    List<Object> vall = new ArrayList<>();
-                    if (jsonObject.has(AllConstants.OPTIONS)) {
-                        JSONArray conceptsOptions = jsonObject.getJSONArray(AllConstants.OPTIONS);
-                        for (int i = 0; i < conceptsOptions.length(); i++) {
-                            JSONObject option = conceptsOptions.getJSONObject(i);
-                            boolean optionValue = option.optBoolean(VALUE);
-                            if (optionValue) {
-                                option.put(AllConstants.TYPE, type);
-                                option.put(AllConstants.PARENT_ENTITY_ID, jsonObject.getString(OPENMRS_ENTITY_ID));
-                                option.put(KEY, jsonObject.getString(KEY));
-                                if (combineCheckboxOptionValues) {
-                                    vall.add(option.optString(AllConstants.TEXT));
-                                } else { // For options with concepts create an observation for each
-                                    createObservation(e, option, String.valueOf(option.getBoolean(VALUE)), entity);
-                                }
+        String entity =  getString(jsonObject, OPENMRS_ENTITY);;
```

Comment:
This should be checked at options level and not at the checkbox level. Usually the options are mapped to concepts

Reply:
Done.